src/components/Calculator.js

```diff
@@ -1,90 +1,135 @@
+/* eslint-disable react/forbid-prop-types */
 import React from 'react';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import 'bootstrap/dist/js/bootstrap.min';
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
+export default class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {
-      result: 0,
-    };
+    this.display = this.display.bind(this);
+    this.calculate = this.calculate.bind(this);
+  }
+
+  display() {
+    const { element } = this.props;
+    const { next } = element;
+    const { operation } = element;
+    const { total } = element;
+    let result = null;
+
+    if (next === null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation}`;
+      } else {
+        result = total;
+      }
+    } else if (next !== null && total === null) {
+      result = next;
+    } else if (next !== null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation} ${next}`;
+      } else {
+        result = total;
+      }
+    } else if (total === null && next === null) {
+      result = 0;
+    }
+    return result;
+  }
+
+  calculate(object, button) {
+    const { updateState } = this.props;
+    let recentObject;
+    try {
+      recentObject = calculate(object, button);
+    } catch {
+      recentObject = {
+        total: 'Module operation by 0 cannot be executed',
+      };
+    }
+
+    this.element = recentObject;
+    updateState(recentObject);
   }
 
   render() {
-    const { result } = this.state;
+    const { element } = this.props;
     return (
       <div className="container" id="calculator">
         <div className="row d-flex justify-content-center">
           <div className="col-4">
             <div className="row mt-5">
               <div className="col-12 text-end display-result px-1 border  border-1">
-                <p className="py-3 mb-0 fs-4 ">{ result }</p>
+                <p className="py-3 mb-0 fs-4 ">{ this.display() }</p>
               </div>
             </div>
             <div className="row">
-              <button type="button" className="col-3 d-flex justify-content-center align-items-center py-3 border  border-1">
+              <button type="button" className="col-3 d-flex justify-content-center align-items-center py-3 border  border-1" onClick={() => { this.calculate(element, 'AC'); }}>
```

Comment:
- [OPTIONAL] There is currently a lot of repetition of the buttons in the render. You can fix this by looping through all of them and then have a <Button/> component that you will pass all the props you want. This will ensure that we adhere to the dry rule